crates/core_simd/src/fmt.rs

```diff
@@ -1,39 +1,23 @@
 use crate::simd::{LaneCount, Simd, SimdElement, SupportedLaneCount};
 use core::fmt;
 
-macro_rules! impl_fmt_trait {
-    { $($trait:ident,)* } => {
-        $(
-            impl<T, const LANES: usize> fmt::$trait for Simd<T, LANES>
-            where
-                LaneCount<LANES>: SupportedLaneCount,
-                T: SimdElement + fmt::$trait,
-            {
-                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                    #[repr(transparent)]
-                    struct Wrapper<'a, T: fmt::$trait>(&'a T);
+impl<T, const LANES: usize> fmt::Debug for Simd<T, LANES>
+where
+    LaneCount<LANES>: SupportedLaneCount,
+    T: SimdElement + fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        #[repr(transparent)]
+        struct Wrapper<'a, T: fmt::Debug>(&'a T);
```

Comment:
Can't all of this just be replaced by `<[T]>::fmt(self.as_array(), f)`?

Reply:
What about simply `Debug::fmt`?

Reply:
@calebzulawski My hope is that forwarding to `[T]`'s formatter reduces overall footprint slightly, and that it should be fine since a test validates the equality. Maybe I will be proven wrong.

Reply:
Makes sense! Maybe `as_slice` would be a little more clear, in that case

Reply:
There is no `Simd::as_slice()`

Reply:
Whoops, true